### PR TITLE
Voice Broadcast - only send a notification on the first chunk

### LINKED ISSUE #7845: Ignore the voice broadcast chunks at the notifications level
#### What would you like to do?
During the notifications handling, we should notify only the first voice chunk of a voice broadcast, and ignore the other events related to the voice broadcast chunks

#### Why would you like to do it?
These events are not displayed in the room timeline, they should not be taken into account in the notifications count.
They are voice messages marked as voice broadcast chunks (they refer to a voice message state event)

There is no case where the user may want to be notified on these chunks, they are supposed to be invisible for them. That is why we can hardcode this mechanism client side without adding config in push rules

#### How would you like to achieve it?
We have to ignore them locally, because these events are encrypted, we could not ignore them server side

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by Florian14**
## Type of change

- [x] WIP Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

During the handling of the notifications, notify only the first voice chunk of a voice broadcast, and ignore the other events related to the voice broadcast chunks

## Motivation and context

Close #7845 

## Screenshots / GIFs

https://user-images.githubusercontent.com/11990514/212134974-33339bae-49ca-40d7-b801-c58ee13231a2.mp4

## Tests

- Enable all notifications on Android device
- Start a voice broadcast from another device
- Pause or wait 2 min to send a chunk
- Verify that a voice broadcast notification appears on Android device
- Resume & Pause again the VB or wait for a second chunk
- Verify that there is no new notification
- Verify that the other notifications (text or simple voice message) are still working

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/notifications/FilteredEventDetector.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * Returns true if the given event should be ignored.
+     * Used to skip notifications if a non expected message is received.
+     */
+    fun shouldBeIgnored(notifiableEvent: NotifiableEvent): Boolean {
+        val session = activeSessionDataSource.currentValue?.orNull() ?: return false
+
+        if (notifiableEvent is NotifiableMessageEvent) {
+            val room = session.getRoom(notifiableEvent.roomId) ?: return false
+            val timelineEvent = room.getTimelineEvent(notifiableEvent.eventId) ?: return false
```
Comment (jonnyandrew) [inline vector/src/main/java/im/vector/app/features/notifications/FilteredEventDetector.kt:41]: If each of these might be null and fall back to returning false, do we risk dropping notifications unnecessarily?

Would it be safer to fall back to returning true?

**Review by jonnyandrew — COMMENTED** [review]
Looks good! I have one concern about the general impact on notifications.

**Review by jonnyandrew — APPROVED** [review]

File: vector/src/main/java/im/vector/app/features/notifications/FilteredEventDetector.kt (hunk 1 above)
Comment (Florian14) [reply vector/src/main/java/im/vector/app/features/notifications/FilteredEventDetector.kt:41]: I think this is fine, returning false means **not** ignore the notification, so it should not drop the notification in this case, wdyt?

Comment (Florian14) [reply vector/src/main/java/im/vector/app/features/notifications/FilteredEventDetector.kt:41]: I did the same as https://github.com/vector-im/element-android/blob/f4b7161db216991a94f1a93a2391d77511fbbf7f/vector/src/main/java/im/vector/app/features/notifications/OutdatedEventDetector.kt#L31, in this detector, messages are not considered as outdated if the room or session is not retrieved

Comment (jonnyandrew) [reply vector/src/main/java/im/vector/app/features/notifications/FilteredEventDetector.kt:41]: That makes sense - I misread the method name. I've edited my comment and approved

**Pull request merged**